### PR TITLE
Update packages, update healthchecks

### PR DESCRIPTION
**Pull request by KostinP**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/251

### CODE REVIEW

File: src/TimeService/Startup.cs [hunk 1]
```diff
@@ -89,11 +99,11 @@ private void ConfigureRepositories(IServiceCollection services)
             services.AddTransient<IWorkTimeRepository, WorkTimeRepository>();
         }
 
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
         {
             app.UseHealthChecks("/api/healthcheck");
```
Comment (borovHokage) [inline src/TimeService/Startup.cs:104]: just a question, should we leave this line? because in others pull requests we deleted it

Comment (KostinP) [reply src/TimeService/Startup.cs]: Fixed.

File: src/TimeService/Startup.cs [hunk 2]
```diff
@@ -17,28 +18,47 @@
...
-            services.AddHealthChecks();
+            services.Configure<BaseRabbitMqOptions>(Configuration.GetSection(BaseRabbitMqOptions.RabbitMqSectionName));
+
+            services.AddHttpContextAccessor();
+
+            services.AddKernelExtensions();
 
             string connStr = Environment.GetEnvironmentVariable("ConnectionString");
+
+            // TODO: Fix HealthChecks work with RabbitMQ
```
Comment (BelehovEgor) [inline src/TimeService/Startup.cs:57]: you already can do it now

File: src/TimeService/TimeService.csproj [hunk 3]
```diff
@@ -2,6 +2,12 @@
 
   <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
+  <ItemGroup>
+    <PackageReference Include="AspNetCore.HealthChecks.Rabbitmq" Version="5.0.1" />
```
Comment (BelehovEgor) [inline src/TimeService/TimeService.csproj:6]: after update hc, delete this

**Review by BelehovEgor — CHANGES_REQUESTED** [review]

File: src/TimeService/Startup.cs [hunk 4]
```diff
@@ -60,6 +80,44 @@ public void ConfigureServices(IServiceCollection services)
...
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+
+                endpoints.MapHealthChecks($"/{_rabbitMqConfig.Password}/hc", new HealthCheckOptions
+                {
+                    Predicate = _ => true,
+                    ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
+                });
```
Comment (djino97) [inline src/TimeService/Startup.cs:113]: Rework

**Review by djino97 — CHANGES_REQUESTED** [review]

File: src/TimeService/Startup.cs (hunk 2 above)
Comment (KostinP) [reply src/TimeService/Startup.cs]: Updated.

File: src/TimeService/TimeService.csproj (hunk 3 above)
Comment (KostinP) [reply src/TimeService/TimeService.csproj]: Updated.

File: Common.Configurations.targets [hunk 5]
```diff
@@ -33,12 +33,12 @@
   </ItemGroup>
 
   <ItemGroup Condition="$(AssemblyName.Contains('.UnitTests')) != 'true'">
-    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.8" />
+    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.14" />
```
[the remark this answers is not in the document]
Comment (KostinP) [reply Common.Configurations.targets]: Updated.

File: src/TimeService/Startup.cs (hunk 4 above)
Comment (KostinP) [reply src/TimeService/Startup.cs]: Updated.

File: src/TimeService.Data.Provider/IDataProvider.cs [hunk 6]
```diff
@@ -1,9 +1,12 @@
-﻿using LT.DigitalOffice.Kernel.Database;
+﻿using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.Kernel.Database;
+using LT.DigitalOffice.Kernel.Enums;
 using LT.DigitalOffice.TimeService.Models.Db;
 using Microsoft.EntityFrameworkCore;
 
 namespace LT.DigitalOffice.TimeService.Data.Provider
 {
+    [AutoInject(InjectType.Scoped)]
```
Comment (TIove) [inline src/TimeService.Data.Provider/IDataProvider.cs:9]: That was Transient before, are you sure that Scoped is better?

Comment (KostinP) [reply src/TimeService.Data.Provider/IDataProvider.cs:9]: Yes

**Review by TIove — APPROVED** [review]

File: test/TimeService.Business.UnitTests/CreateLeaveTimeCommandTests.cs [hunk 7]
```diff
@@ -1,22 +1,17 @@
...
-using System;
-using System.Collections.Generic;
 
 namespace LT.DigitalOffice.TimeService.Business.UnitTests
 {
     public class CreateLeaveTimeCommandTests
     {
-        private Mock<IValidator<CreateLeaveTimeRequest>> validatorMock;
-        private Mock<IMapper<CreateLeaveTimeRequest, DbLeaveTime>> mapperMock;
+        private Mock<ICreateLeaveTimeRequestValidator> validatorMock;
```
Comment (BelehovEgor) [inline test/TimeService.Business.UnitTests/CreateLeaveTimeCommandTests.cs:13]: use _ in private fields

File: test/TimeService.Mappers.UnitTests/CreateWorkTimeMapperTests.cs [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        [Test]
+        public void ShouldReturnDbWorkTimeWhenMappingValidCreateWorkTimeRequestWithNullDescription()
+        {
+            _createRequest.Description = null;
+            _expectedDbWorkTimeWithoutId.Description = null;
+
+            var newWortTime = _createRequestMapper.Map(_createRequest);
+            _expectedDbWorkTimeWithoutId.Id = newWortTime.Id;
+
+            Assert.IsInstanceOf<Guid>(newWortTime.Id);
```
Comment (BelehovEgor) [inline test/TimeService.Mappers.UnitTests/CreateWorkTimeMapperTests.cs:73]: what else can there be besides Guid?

**Review by BelehovEgor — CHANGES_REQUESTED** [review]

File: src/TimeService.Validation/CreateWorkTimeRequestValidator.cs [hunk 9]
```diff
@@ -2,14 +2,15 @@
...
+using LT.DigitalOffice.TimeService.Validation.Interfaces;
 using Microsoft.AspNetCore.Mvc;
 using System;
 using System.Globalization;
 using System.Linq;
 
 namespace LT.DigitalOffice.TimeService.Validation
 {
-    public class CreateWorkTimeRequestValidator : AbstractValidator<CreateWorkTimeRequest>
+    public class CreateWorkTimeRequestValidator : AbstractValidator<CreateWorkTimeRequest>, ICreateWorkTimeRequestValidator
```
[the remark this answers is not in the document]
Comment (KostinP) [reply src/TimeService.Validation/CreateWorkTimeRequestValidator.cs:13]: [AutoInject]

File: test/TimeService.Mappers.UnitTests/CreateWorkTimeMapperTests.cs (hunk 8 above)
Comment (KostinP) [reply test/TimeService.Mappers.UnitTests/CreateWorkTimeMapperTests.cs:73]: Thrown if value is null or expectedType is not in the inheritance hierarchy of value.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.assert.isinstanceoftype?redirectedfrom=MSDN&view=mstest-net-1.3.2#Microsoft_VisualStudio_TestTools_UnitTesting_Assert_IsInstanceOfType_System_Object_System_Type_System_String_System_Object___

File: test/TimeService.Business.UnitTests/CreateLeaveTimeCommandTests.cs (hunk 7 above)
Comment (KostinP) [reply test/TimeService.Business.UnitTests/CreateLeaveTimeCommandTests.cs]: Fixed

File: src/TimeService/Startup.cs [hunk 10]
```diff
@@ -1,44 +1,65 @@
...
+
+            _rabbitMqConfig = Configuration
+                .GetSection(BaseRabbitMqConfig.SectionName)
+                .Get<RabbitMqConfig>();
+
+            _serviceInfoConfig = Configuration
+                .GetSection(BaseServiceInfoConfig.SectionName)
+                .Get<BaseServiceInfoConfig>();
+
+            using var loggerFactory = LoggerFactory.Create(builder =>
```
[the remark this answers is not in the document]
Comment (KostinP) [reply src/TimeService/Startup.cs:40]: Syntactic sugar so as not to be wrapped in using

File: test/TimeService.Mappers.UnitTests/CreateWorkTimeMapperTests.cs (hunk 8 above)
Comment (BelehovEgor) [reply test/TimeService.Mappers.UnitTests/CreateWorkTimeMapperTests.cs:73]: But you use Guid, not Guid?, this type can't be null

File: src/TimeService/Startup.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/TimeService/Startup.cs:40]: ok

File: test/TimeService.Mappers.UnitTests/CreateWorkTimeMapperTests.cs (hunk 8 above)
Comment (KostinP) [reply test/TimeService.Mappers.UnitTests/CreateWorkTimeMapperTests.cs:72]: Removed

File: src/TimeService.Business/CreateWorkTimeCommand.cs [hunk 11]
```diff
@@ -12,23 +14,23 @@ namespace LT.DigitalOffice.TimeService.Business
...
+        private readonly ICreateWorkTimeRequestValidator _validator;
+        private readonly ICreateWorkTimeMapper _mapper;
+        private readonly IWorkTimeRepository _repository;
 
         public CreateWorkTimeCommand(
-            [FromServices] IValidator<CreateWorkTimeRequest> validator,
-            [FromServices] IMapper<CreateWorkTimeRequest, DbWorkTime> mapper,
+            [FromServices] ICreateWorkTimeRequestValidator validator,
+            [FromServices] ICreateWorkTimeMapper mapper,
             [FromServices] IWorkTimeRepository repository)
```
Comment (djino97) [inline src/TimeService.Business/CreateWorkTimeCommand.cs:24]: remove FromService

File: src/TimeService.Mappers/Requests/Interfaces/IEditWorkTimeMapper.cs [hunk 12]
```diff
@@ -0,0 +1,12 @@
+﻿using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.TimeService.Mappers.Interfaces;
+using LT.DigitalOffice.TimeService.Models.Db;
+using LT.DigitalOffice.TimeService.Models.Dto;
+
+namespace LT.DigitalOffice.TimeService.Mappers.Requests.Interfaces
+{
+    [AutoInject]
+    public interface IEditWorkTimeMapper : IMapper<EditWorkTimeRequest, DbWorkTime>
```
Comment (24eeeee) [inline src/TimeService.Mappers/Requests/Interfaces/IEditWorkTimeMapper.cs:9]: As I know, we considered to name mapper after entity it returns. Can you, please, rename them all?

**Review by djino97 — APPROVED** [review]

**Review by djino97 — APPROVED** [review]

**Review by 24eeeee — APPROVED** [review]

File: src/TimeService.Mappers/Requests/Interfaces/IEditWorkTimeMapper.cs (hunk 12 above)
Comment (KostinP) [reply src/TimeService.Mappers/Requests/Interfaces/IEditWorkTimeMapper.cs:9]: Renamed

File: src/TimeService.Business/CreateWorkTimeCommand.cs (hunk 11 above)
Comment (KostinP) [reply src/TimeService.Business/CreateWorkTimeCommand.cs]: Removed.

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**